frontend/public/components/catalog.jsx

```diff
@@ -0,0 +1,220 @@
+        const tileIconClass = getImageStreamTagAnnotation(imagestream, 'iconClass', builderSpecTagName);
+        const tileDescription = getImageStreamTagAnnotation(imagestream, 'description', builderSpecTagName);
+        const tileProvider = getImageStreamTagAnnotation(imagestream, 'openshift.io/provider-display-name', builderSpecTagName);
+        return {
+          resource: imagestream,
+          kind,
+          tileName,
+          tileIconClass,
+          tileDescription,
+          tileProvider
+        };
+      }
+      return {
+        hidden: true
+      };
+    });
+
+    return _.filter(imagestreams, imagestream => {
+      return !imagestream.hidden;
+    });
+  }
+
+  handleFilterChange(event) {
+    this.setState({filterValue: event.target.value});
+  }
+
+  clearFilter() {
+    this.setState({filterValue: ''});
+  }
+
+  render() {
+    const {loaded, loadError} = this.props;
+    const {filteredItems} = this.state;
+
+    return <div className="co-m-pane">
+      <div className="co-m-pane__body">
+        <StatusBox data={filteredItems} loaded={loaded} loadError={loadError} label="Resources">
+          <CatalogList items={filteredItems} />
+        </StatusBox>
+      </div>
+    </div>;
+  }
+}
+
+CatalogListPage.displayName = 'CatalogList';
+
+CatalogListPage.propTypes = {
+  obj: PropTypes.object
+};
+
+// eventually may use namespace
+// eslint-disable-next-line no-unused-vars
+export const Catalog = ({namespace, flags}) => {
+  const resources = [];
+  if (flags.SERVICE_CATALOG) {
+    resources.push({
+      isList: true,
+      kind: 'ClusterServiceClass',
+      namespaced: false,
+      prop: 'clusterserviceclasses'
+    });
+  }
+  if (flags.OPENSHIFT) {
+    resources.push({
+      isList: true,
+      kind: 'ImageStream',
+      namespace: 'openshift',
+      prop: 'imagestreams'
+    });
+  }
+  return <div className="catalog">
+    <Firehose resources={resources}>
+      <CatalogListPage />
+    </Firehose>
+  </div>;
+};
+
+Catalog.displayName = 'Catalog';
+
+Catalog.propTypes = {
+  namespace: PropTypes.string.isRequired,
+  flags: PropTypes.string.isRequired
+};
+
+
+class CatalogPage_ extends React.PureComponent {
+  render() {
+    const {match, flags} = this.props;
+    const namespace = _.get(match, 'params.ns');
+
+    if (flagPending(flags.OPENSHIFT) || flagPending(flags.SERVICE_CATALOG)) {
+      return null;
+    }
+
+    return <React.Fragment>
+      <Helmet>
+        <title>Catalog</title>
+      </Helmet>
+      <NavTitle title="Catalog" />
+      <Catalog
+        namespace={namespace}
+        flags={flags}
+      />
+    </React.Fragment>;
+  }
+}
+
+CatalogPage_.displayName = 'CatalogPage';
+
+CatalogPage_.propTypes = {
+  match: PropTypes.object.isRequired,
+  flags: PropTypes.object.isRequired
+};
+
+// TODO: do we need to check for OPENSHIFT flag for catalog?
+export const CatalogPage = connectToFlags(FLAGS.OPENSHIFT, FLAGS.SERVICE_CATALOG)(CatalogPage_);
```

Comment:
We should probably connect Catalog to flags, not CatalogPage

Reply:
Sorry, I don't follow?

Reply:
Move connectToFlags to the child `Catalog` component, not `CatalogPage`

Reply:
fixed